tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/PopulateBasicPropertiesHeadersIntegration.cs

```diff
@@ -45,7 +51,29 @@ public class PopulateBasicPropertiesHeadersIntegration
             return new CallTargetReturn<TReturn?>(returnValue);
         }
 
-        returnValue ??= CachedBasicPropertiesHelper<TReturn>.CreateHeaders();
+        // PopulateBasicPropertiesHeaders returns null if the supplied IReadOnlyBasicProperties
+        // does not have to be modified or if it's a writable instance.
+        // If that is the case then we have to fetch IReadOnlyBasicProperties from the argument
+        // list instead of creating a new instance that overwrites the supplied properties.
+        if (returnValue is null)
+        {
+            if (state.State.DuckIs<IBasicProperties>())
+            {
+                // Use the existing basic properties if possible...
+                returnValue = (TReturn)state.State!;
+            }
+            else if (state.State.DuckIs<IReadOnlyBasicProperties>())
+            {
+                // if not create new instance using the copy constructor on BasicProperties.
+                returnValue = CachedBasicPropertiesHelper<TReturn>.CreateHeaders(state.State!);
+            }
+            else
+            {
+                // This case cannot happen as argument is not nullable.
```

Comment:
If it shouldn't happen, maybe we should add an error log for when it inevitably does 😉